x/protorev/keeper/posthandler.go

```diff
@@ -63,6 +61,9 @@ func (protoRevDec ProtoRevDecorator) AnteHandle(ctx sdk.Context, tx sdk.Tx, simu
 		ctx.Logger().Error("ProtoRevTrade failed with error", err)
 	}
 
+	// Delete swaps to backrun for next transaction without consuming gas
+	protoRevDec.ProtoRevKeeper.DeleteSwapsToBackrun(ctx.WithGasMeter(upperGasLimitMeter))
```

Comment:
The gas is still consumed, correct? Did you mean that the limit is refreshed?

Reply:
The gas is not consumed from the transaction's gas meter, but the new gas meter passed in (which is only used temporarily during the function call), added clarifying comment here: https://github.com/osmosis-labs/osmosis/pull/5045/commits/61d3474a6a489d1247c556950e28aaa46209ebb0

This was made this way (as opposed to using the gas meter in cacheCtx) because no matter what happens with the backrun itself (whether or not it errors), we want to clear out the swaps to start fresh.